rasa/nlu/components.py

```diff
@@ -322,6 +341,70 @@ def validate_required_components_from_data(
         )
 
 
+def warn_of_competing_extractors(pipeline: List["Component"]) -> None:
+    """Warns the user when using competing extractors.
+
+    Competing extractors are e.g. `CRFEntityExtractor` and `DIETClassifier`.
+    Both of these look for the same entities based on the same training data
+    leading to ambiguity in the results.
+    """
+    general_purpose_entity_extractors = [
+        "DIETClassifier",
+        "CRFEntityExtractor",
+        "MitieEntityExtractor",
+    ]
+    extractors_in_pipeline = find_components_in_pipeline(
+        general_purpose_entity_extractors, pipeline
+    )
+    if len(extractors_in_pipeline) > 1:
+        rasa.shared.utils.io.raise_warning(
+            f"You have defined multiple entity extractors that do the same job"
+            f"in your pipeline: "
+            f"{', '.join(extractors_in_pipeline)}. This can lead to the same entity getting"
+            f"extracted multiple times. Please read the documentation section"
+            f"on entity extractors to make sure you understand the implications:"
+            f"{DOCS_URL_COMPONENTS}#entity-extractors"
+        )
+
+
+def warn_of_competition_with_regex_extractor(
+    pipeline: List["Component"], data: TrainingData
+) -> None:
+    """Warns when regex entity extractor is competing with a general one.
+
+    This might be the case when the following conditions are all met:
+    * You are using a general entity extractor and the `RegexEntityExtractor`
+    * AND you have regex patterns for entity type A
+    * AND you have annotated text examples for entity type A
+    """
+    general_entity_extractors = [
+        "DIETClassifier",
+        "CRFEntityExtractor",
+        "MitieEntityExtractor",
+    ]
+    present_general_extractors = find_components_in_pipeline(
+        general_entity_extractors, pipeline
+    )
+    has_general_extractors = len(present_general_extractors) > 0
+    has_regex_extractor = any_components_in_pipeline(["RegexEntityExtractor"], pipeline)
+
+    regex_entity_types = {rf["name"] for rf in data.regex_features}
+    overlap_between_types = data.entities.intersection(regex_entity_types)
+    has_overlap = len(overlap_between_types) > 0
+
+    if has_general_extractors and has_regex_extractor and has_overlap:
```

Comment:
isn't this check enough?
```suggestion
    if has_overlap:
```

Reply:
Then it could for example trigger if you have no regex extractor but regex features. Which would be a false positive. Or say you have the regex extractor but no general extractor.